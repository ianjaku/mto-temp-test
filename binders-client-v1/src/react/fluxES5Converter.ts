/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
