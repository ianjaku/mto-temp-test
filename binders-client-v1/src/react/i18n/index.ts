import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
