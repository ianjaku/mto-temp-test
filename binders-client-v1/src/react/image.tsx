/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
