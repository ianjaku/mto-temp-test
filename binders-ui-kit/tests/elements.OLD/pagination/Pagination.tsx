/* eslint-disable @typescript-eslint/no-unused-vars */
import * as React from "react";
import Pagination from "../../../src/elements/pagination";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";


const paginationLongRange = wrapWithTheme(<Pagination max={500} />);
const paginationDisplayPages = wrapWithTheme(<Pagination max={500} displayPages={20} />);

const renderPaginationWithCallback = c => wrapWithTheme(<Pagination max={500} onPageChange={c} />);

// most preferred
describe("Pagination", () => {
    test("Pagination Long range (snapshot)", () => {
        const modal = create(paginationLongRange);
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Pagination Display pages (snapshot)", () => {
        const modal = create(paginationDisplayPages);
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    // test last
    test("Page texts should be correct", () => {
        const element = mount(paginationLongRange);
        const amountFirstPage = element.find(".pagination > li").length;
        const firstPageText = element
            .find(".pagination > li")
            .at(0)
            .text();
        const previousPageText = element
            .find(".pagination > li")
            .at(1)
            .text();
        const page1Text = element
            .find(".pagination > li")
            .at(2)
            .text();
        const page2Text = element
            .find(".pagination > li")
            .at(3)
            .text();
        const pageNextPage = element
            .find(".pagination > li")
            .at(7)
            .text();
        const pageLastPage = element
            .find(".pagination > li")
            .at(8)
            .text();

        expect(amountFirstPage).toEqual(9);
        // todo: fix the names
        // expect(firstPageText).toEqual("keyboard_arrow_leftkeyboard_arrow_leftFirst");
        // expect(previousPageText).toEqual("keyboard_arrow_leftPrevious");
        expect(page1Text).toEqual("1");
        expect(page2Text).toEqual("2");
        // expect(pageNextPage).toEqual("Nextkeyboard_arrow_right");
        // expect(pageLastPage).toEqual("Lastkeyboard_arrow_rightkeyboard_arrow_right");
    });

    // test first and previous
    test("First and Previous should be disabled on page 1", () => {
        const element = mount(paginationLongRange);
        const firstPageAttr = element
            .find(".pagination > li")
            .at(0)
            .hasClass("disabled");
        const previousPageAttr = element
            .find(".pagination > li")
            .at(1)
            .hasClass("disabled");

        expect(firstPageAttr).toEqual(true);
        expect(previousPageAttr).toEqual(true);
    });

    test("First should be disabled and Previous should be visible on page 2", () => {
        const mockCallback = jest.fn();

        const element = mount(renderPaginationWithCallback(mockCallback));
        const page2Element = element.find(".pagination > li").at(3);

        page2Element.simulate("click");
        let firstPageElement = element.find(".pagination > li").at(0);
        let previousPageElement = element.find(".pagination > li").at(1);
        const pageNextElement = element.find(".pagination > li").at(7);

        expect(mockCallback.mock.calls.length).toEqual(1);
        expect(firstPageElement.hasClass("disabled")).toEqual(true);
        expect(previousPageElement.hasClass("disabled")).toEqual(false);

        pageNextElement.simulate("click");
        firstPageElement = element.find(".pagination > li").at(0);
        previousPageElement = element.find(".pagination > li").at(1);
        expect(mockCallback.mock.calls.length).toEqual(2);
        expect(firstPageElement.hasClass("disabled")).toEqual(false);
        expect(previousPageElement.hasClass("disabled")).toEqual(false);
    });

    test("Previous and last should not be visible when on last page", () => {
        const mockCallback = jest.fn();
        const element = mount(renderPaginationWithCallback(mockCallback));

        let pageNextElement = element.find(".pagination > li").at(7);
        let pageLastElement = element.find(".pagination > li").at(8);

        expect(mockCallback.mock.calls.length).toEqual(0);
        expect(pageLastElement.hasClass("disabled")).toEqual(false);
        expect(pageNextElement.hasClass("disabled")).toEqual(false);

        // go to last page
        pageLastElement.simulate("click");
        pageNextElement = element.find(".pagination > li").at(7);
        pageLastElement = element.find(".pagination > li").at(8);

        expect(mockCallback.mock.calls.length).toEqual(1);
        expect(pageLastElement.hasClass("disabled")).toEqual(true);
        expect(pageNextElement.hasClass("disabled")).toEqual(true);

        // go to previous page
        element
            .find(".pagination > li")
            .at(1)
            .simulate("click");
        pageNextElement = element.find(".pagination > li").at(7);
        pageLastElement = element.find(".pagination > li").at(8);

        expect(mockCallback.mock.calls.length).toEqual(2);
        expect(pageLastElement.hasClass("disabled")).toEqual(true);
        expect(pageNextElement.hasClass("disabled")).toEqual(false);

        // go 1 more page back
        element
            .find(".pagination > li")
            .at(1)
            .simulate("click");
        pageNextElement = element.find(".pagination > li").at(7);
        pageLastElement = element.find(".pagination > li").at(8);

        expect(mockCallback.mock.calls.length).toEqual(3);
        expect(pageLastElement.hasClass("disabled")).toEqual(false);
        expect(pageNextElement.hasClass("disabled")).toEqual(false);
    });
});
