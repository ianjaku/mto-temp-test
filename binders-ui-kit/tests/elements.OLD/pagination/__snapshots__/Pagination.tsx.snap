// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Pagination Display pages (snapshot) 1`] = `
<ul
  className="pagination"
>
  <span>
    Page
  </span>
  <li
    className="disabled"
    onClick={undefined}
  >
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_left
    </span>
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_left
    </span>
    First
  </li>
  <li
    className="disabled"
    onClick={undefined}
  >
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_left
    </span>
    Previous
  </li>
  <li
    className="active"
    onClick={[Function]}
  >
    1
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    2
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    3
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    4
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    5
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    6
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    7
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    8
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    9
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    10
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    11
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    12
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    13
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    14
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    15
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    16
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    17
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    18
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    19
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    20
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    21
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Next 
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_right
    </span>
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Last
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_right
    </span>
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_right
    </span>
  </li>
</ul>
`;

exports[`Pagination Pagination Long range (snapshot) 1`] = `
<ul
  className="pagination"
>
  <span>
    Page
  </span>
  <li
    className="disabled"
    onClick={undefined}
  >
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_left
    </span>
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_left
    </span>
    First
  </li>
  <li
    className="disabled"
    onClick={undefined}
  >
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_left
    </span>
    Previous
  </li>
  <li
    className="active"
    onClick={[Function]}
  >
    1
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    2
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    3
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    4
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    5
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Next 
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_right
    </span>
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    Last
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_right
    </span>
    <span
      className="material-icons pagination-icon"
      color=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      keyboard_arrow_right
    </span>
  </li>
</ul>
`;
